### DISCUSSION TITLE
Welcome to MQL5_CSharp Discussions!

### DISCUSSION THREAD

**Discussion by darthlinuxer** [darthlinuxer/MQL5_CSharp#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by JoelBoily** [comment]
Hi bro. First thank you for your MQL5Csharp.
I use it to control my EA. All is working fine. I use buttons to send orders and i also fetch double arrays from mt5. I display these values in my panel. Using a button event i can "refresh" the values. But i tried to add a timer to "refresh" but it never work. I read some infos about managing a timer for a DLL but i not succeed. How can i replace "refresh button" with a timer in your code ? Can you help me? What is the best way to update the display of the values? Timer or comparing if values changed?![20240407_180902.jpg](https://github.com/darthlinuxer/MQL5_CSharp/assets/108032096/10905583-130f-4167-a3cd-3f4ee1044243)

**Reply by JoelBoily** [reply]
Thank you again doing this MQL5CSHARP... i wiped out my old panel and i begin to made a full flanged one. Your work is appreciated in Canada. My name is Joel and trading is my passion.

**Reply by darthlinuxer** [reply]
Will download the Repo and install MQL5 to continue the progress on this project sometime this week . Your idea might help others.

**Reply by JoelBoily** [reply]
I trully appreciate your help. I was blocked and you help me. Yeaahh !!! thank you very very much !!!!

**Reply by darthlinuxer** [reply]
Hi Joel, I´ve added your refresh timer to the .NET 4 Framework project.

**Reply by JoelBoily** [reply]
Hi Camilo !!! I thank you again ... i will look at it after my job when i will be at home ...

**Reply by JoelBoily** [reply]
![Screen Shot 04-08-24 at 07 00 PM](https://github.com/darthlinuxer/MQL5_CSharp/assets/108032096/2c632aea-21f6-4573-81a2-f33f45916aa1)
I modified my code based on your commit and it is working bro !!! I just did a small test with account profit. Now i can begin to send my data that i had in my old panel .... i was not far yesterday ... my code missed --> this.components = new System.ComponentModel.Container(); ...... this.timer1 = new System.Windows.Forms.Timer(this.components); I was not passing this.component to my new timer ... i am newby in C# ... i mostly code MQL5 ... A big thank you for your help ... i appreciate it ... you made me smilling :)

**Reply by JoelBoily** [reply]
Bro , do you think that we can use WPF instead Windows Form?

**Comment by darthlinuxer** [comment]
Yes it is possible. Will create a template for that too

Em qua., 10 de abr. de 2024 às 17:55, JoelBoily ***@***.***>
escreveu:

**Reply by JoelBoily** [reply]
You make me smilling again Camilo. :-) i will wait before to invest time doing the panel in WF. WPF will be more scalable in the future. You do not imagine how big i appreciate your work. I would like to have your knowledge.

**Comment by mackferson67** [comment]
Hello Camilo, thanks for sharing your project.
Indeed, your approach is simpler than our colleague's, but in both projects I have a problem integrating the Form with MT5 that causes the application to unexpectedly close.
In the case of your Script, when I use the CLOSE button, sometimes (rarely) the form closes and the script leaves the MT5 window and the process list. Most of the time, the Script hangs and MT5 closes unexpectedly.
Are there any integration details that must be corrected to eliminate this instability when we want to close the form without taking down all other EAs and Indicators?
Thanks in advance.
PS. If you have anyone you know in RS, I hope you are all well.
